examples/nlp/active_learning_review_classification.py

```diff
@@ -0,0 +1,395 @@
+test_dataset = (
+    zeros_dataset.skip(val_split)
+    .take(test_split)
+    .concatenate(ones_dataset.skip(val_split).take(test_split))
+)
+train_dataset = (
+    zeros_dataset.skip(val_split + test_split)
+    .take(train_samples)
+    .concatenate(ones_dataset.skip(val_split + test_split).take(train_samples))
+)
+
+print(f"Initial training set size: {train_samples*2}")
+print(f"Validation set size: {val_split*2}")
+print(f"Testing set size: {test_split*2}")
+
+# Dumping the remaining samples in their respective data pools
+pool_zeros = zeros_dataset.skip(val_split + test_split + train_samples)
+pool_ones = ones_dataset.skip(val_split + test_split + train_samples)
+
+print(f"Unlabeled zeros pool: {25000-val_split-test_split-train_samples}")
+print(f"Unlabeled ones pool: {25000-val_split-test_split-train_samples}")
+
+"""
+### Fitting the TextVectorization layer
+
+Since we are working with textual data, we will need to encode the text as vectors which
+would then be passed through an Embedding layer. To make this tokenization process
+faster, we use the `map()` function with it's parallelization functionality.
+"""
+
+vectorizer = layers.TextVectorization(3000, output_sequence_length=150,)
+vectorizer.adapt(
+    train_dataset.map(lambda x, y: x, num_parallel_calls=tf.data.AUTOTUNE).batch(256)
+)
+
+
+def vectorize_text(text, label):
+    text = vectorizer(text)
+    return text, label
+
+
+train_dataset = train_dataset.map(
+    vectorize_text, num_parallel_calls=tf.data.AUTOTUNE
+).prefetch(tf.data.AUTOTUNE)
+pool_zeros = pool_zeros.map(vectorize_text, num_parallel_calls=tf.data.AUTOTUNE)
+pool_ones = pool_ones.map(vectorize_text, num_parallel_calls=tf.data.AUTOTUNE)
+
+val_dataset = val_dataset.batch(256).map(
+    vectorize_text, num_parallel_calls=tf.data.AUTOTUNE
+)
+test_dataset = test_dataset.batch(256).map(
+    vectorize_text, num_parallel_calls=tf.data.AUTOTUNE
+)
+
+"""
+## Creating Helper Functions
+"""
+
+# Helper function for appending new history objects with older ones
+def append_history(losses, val_losses, acc, val_acc, history):
+    losses = losses + history.history["loss"]
+    val_losses = val_losses + history.history["val_loss"]
+    acc = acc + history.history["accuracy"]
+    val_acc = val_acc + history.history["val_accuracy"]
+    return losses, val_losses, acc, val_acc
+
+
+# Plotter function
+def plot_history(losses, val_losses, accuracies, val_accuracies):
+    plt.plot(losses)
+    plt.plot(val_losses)
+    plt.legend(["train_loss", "val_loss"])
+    plt.xlabel("Epochs")
+    plt.ylabel("Loss")
+    plt.show()
+
+    plt.plot(accuracies)
+    plt.plot(val_accuracies)
+    plt.legend(["train_accuracy", "val_accuracy"])
+    plt.xlabel("Epochs")
+    plt.ylabel("Accuracy")
+    plt.show()
+
+
+"""
+## Creating the Model
+"""
+
+
+def create_model():
+    model = keras.models.Sequential()
+    model.add(layers.Embedding(3000, 128))
+    model.add(layers.Bidirectional(layers.LSTM(32, return_sequences=True)))
```

Comment:
You might want to use a bigram model instead, which is what works best on IMDB, since it will be much faster.